module/zfs/vdev.c

```diff
@@ -5899,15 +5929,8 @@ vdev_props_set_sync(void *arg, dmu_tx_t *tx)
 	/*
 	 * Set vdev property values in the vdev props mos object.
 	 */
-	if (vd->vdev_root_zap != 0) {
-		objid = vd->vdev_root_zap;
-	} else if (vd->vdev_top_zap != 0) {
-		objid = vd->vdev_top_zap;
-	} else if (vd->vdev_leaf_zap != 0) {
-		objid = vd->vdev_leaf_zap;
-	} else {
+	if (vdev_prop_get_objid(vd, &objid) != 0)
 		panic("unexpected vdev type");
```

Comment:
Would it make sense to move the `panic()` into `vdev_prop_get_objid()` to save having to test it everywhere it is used?

Reply:
We use it in `vdev_prop_get_int`, and then it is used for example [here](https://github.com/openzfs/zfs/blob/master/module/zfs/vdev.c#L6075) to generate an error.

So I would leave it as is for a consumer to decide if it should panic or not.